src/Adapter/Alias/Repository/AliasRepository.php

```diff
@@ -151,4 +138,9 @@ public function getAliasesBySearchTerm(string $searchTerm): array
 
         return $qb->execute()->fetchFirstColumn();
     }
+
+    public function delete(AliasId $aliasId): void
+    {
+        $this->get($aliasId)->delete();
```

Comment:
```suggestion
        $this->deleteObjectModel($this->get($aliasId), ....);
```
You shouldn't use the `ObjectModel::delete` method directly, but instead rely on the `deleteObjectModel` from the abstract helper methods to handle the exception correctly in case of failure You will have to introduce a dedicated `CannotDeleteAliasException`